server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -930,6 +938,29 @@ public void onFailure(Exception e) {
         });
     }
 
+    /**
+     * Adapts failure store resolver function so that if the failure store node feature is not present on every node it reverts to the
+     * old ingest behavior.
+     * @param resolveFailureStore Function that surfaces if failures for an index should be redirected to failure store.
+     * @return An adapted function that mutes the original if the cluster does not have the node feature universally applied.
+     */
+    private Function<String, Boolean> wrapResolverWithFeatureCheck(Function<String, Boolean> resolveFailureStore) {
+        final boolean clusterHasFailureStoreFeature = featureService.clusterHasFeature(
+            clusterService.state(),
+            DataStream.DATA_STREAM_FAILURE_STORE_FEATURE
+        );
+        return (indexName) -> {
+            Boolean resolution = resolveFailureStore.apply(indexName);
+            if (clusterHasFailureStoreFeature || resolution == null) {
```

Comment:
A very minor thing, but you don't need to run the `resolveFailureStore` function if `clusterHashFailureStoreFeature` is false, right?

Reply:
Fair, yeah, I'll move some logic around to cut this down.